src/creature_senses.c

```diff
@@ -812,12 +812,16 @@ TbBool line_of_sight_3d(const struct Coord3d *frpos, const struct Coord3d *topos
     nextpos.y.val = prevpos.y.val + increase_y;
     
     //Z position overshoots, which returns incorrect results. Workaround until a proper fix is made:
```

Comment:
I think this workaround exists because the changed condition I mentioned in my previous comment.

Reply:
I implemented the workaround in 658ab2ab90eeb0fb696d0c80df88f8b4c87f4cb1, but unfortunately changing > to >= does not do anything to fix the original problem.